src/metrics/scrapper.py

```diff
@@ -1,32 +1,121 @@
 # Python Imports
-import requests
-from typing import List
 import logging
+import pandas as pd
+from itertools import chain
+from typing import List, Dict
+from pathlib import Path
 
 # Project Imports
-import src.logging.logger
 from src.metrics import scrape_utils
+from result import Ok, Err, Result
+from src.utils.file_utils import read_yaml_file
+from src.utils.queries import get_query_data
 
 logger = logging.getLogger(__name__)
 
 
 class Scrapper:
-    def __init__(self, url: str, namespace: str, metrics: List):
+    def __init__(self, url: str, query_config_file: str, out_folder: str):
         self._url = url
-        self._namespace = namespace
-        self._metrics = metrics
+        self._query_config_file = query_config_file
+        self._out_folder = out_folder
+        self._set_query_config()
         # TODO make interval match value in cluster
-        self._template = "irate($metric{namespace=$namespace}[3m])"
 
-    def make_queries(self):
-        for metric in self._metrics:
-            query = self._template.replace("$metric", metric)
-            query = query.replace("$namespace", self._namespace)
-            promql = scrape_utils.create_promql(self._url, query, 1, 60)
-            logger.info(f"Promql: {promql}")
-            response = requests.get(promql)
-            logger.info(f"Response: {response.status_code}")
+    def query_and_dump_metrics(self):
+        for metric_dict_item in self._query_config['metrics_to_scrape']:
+            metric, column_name = next(iter(metric_dict_item.items()))
+            logger.info(f'Querying {metric}')
+            promql = self._create_query(metric, self._query_config['scrape_config'])
 
+            match get_query_data(promql):
+                case Ok(data):
+                    logger.info(f'Successfully extracted {metric} data from response')
+                case Err(err):
+                    logger.info(err)
+                    continue
 
+            logger.info(f'Dumping {metric} data to .csv')
+            self._dump_data(metric, column_name, data)
 
+    def _set_query_config(self):
+        self._query_config = read_yaml_file(self._query_config_file)
```

Comment:
It's a good practice to set attributes in the `__init__`, even if it's to `None`. That way you can, at a glimpse, understand all the parts of a class.

Reply:
True, added.

Also, not a specific one right now. Maybe later we would change this to use several files, but right now there is no specific reason.